articles/django-rest-api-mongodb-notes-app/index.md

```diff
@@ -0,0 +1,231 @@
+
+In this post, you will:
+
+- Understand to integrate MongoDB into Django projects
+- Learn how to set up and run Django REST API
+- Create CRUD features in Django REST API
+- Build a Notes app API
+
+## Prerequisites
+
+- Python 3 installed on your machine. You can install Python 3 from the [Python website](https://www.python.org/download/releases/3.0/).
+- Experience with Python and Django
+
+## Setting up MongoDB Server
+
+You may download and install the MongoDB database server from the [official MongoDB website](https://www.mongodb.com/try/download/community).
+
+## Setting up the Virtual Environment
+
+We will install `virtualenv` to enable a virtual environment for our project. It will enable one to isolate a project and its dependencies from other projects on your machine. Run the following command to install `virtualenv`:
+
+```bash
+python -m pip install --user virtualenv
+```
+
+Next, let's create a folder for our project and then create a virtual environment inside it. Use the following command to create a folder called `django_mongodb_project`.
+
+```bash
+mkdir django_mongodb_project
+```
+
+Move into the project folder with the following command:
+
+```bash
+cd django_mongodb_project
+```
+
+Create a virtual environment called `venv` with the following command:
+
+```bash
+virtualenv venv
+```
+
+Next, activate the environment:
+
+```bash
+source venv/bin/activate
+```
+
+If you use Windows, activate the virtual environment with the following command:
+
+```bash
+.\venv\Scripts\activate
+```
+
+## Install Packages
+
+We will need the `django`, `djangorestframework`, and `djongo` packages.
+django: the Django framework package
+djangorestframework: the Django REST Framework for developing APIs in Django.
+djongo: a tool that maps Python objects to MongoDB documents
+
+Let us install Django and the Django REST framework with the following command:
+
+```bash
+pip install django django-rest-framework djongo
+```
+
+Now, create a Django project called `notes_app`:
+
+```bash
+django-admin startproject notes_app
+cd notes_app
+```
+
+Then, create an app called `api` inside the `notes_app` project.
+
+```bash
+django-admin startapp api
+```
+
+Navigate to the `settings.py` file in the project directory. Then, change the `INSTALLED_APPS` list by adding our created app, `api` and `rest_framework`:
+
+```python
+INSTALLED_APPS = [
+    'django.contrib.admin',
+    'django.contrib.auth',
+    'django.contrib.contenttypes',
+    'django.contrib.sessions',
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
+    'rest_framework',
+    'api',
+]
+```
+
+## Configuring MongoDB
+
+Navigate to the `settings.py` file and change the `DATABASES` setting as follows:
+
+```python
+DATABASES = {
+   'default' : {
+      'ENGINE' : 'django_mongodb_engine',
+      'NAME' : 'notes_database'
+   }
+}
```

Comment:
This configurations throw the following error:
```properties
ModuleNotFoundError: No module named 'django_mongodb_engine'
```
Check your database engine. Did you mean `djongo`?

Reply:
Yeah, I used djongo